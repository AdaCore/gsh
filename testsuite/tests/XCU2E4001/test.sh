echo "++ ensure that a variable set in a sourced script is visible"
. ./test.sub
echo "return status: $?"
echo "\$toto='$toto'"

echo "++ use of source builtin without any argument"
. 2>/dev/null
echo "return status: $?"

echo "++ sourcing inexistant script"
. ./foo.sub 2>/dev/null
echo "return status: $?"

echo "++ sourcing a script + positional arg"
. ./test.sub a b c
echo "return status: $?"

echo "++ return in a sourced scripts"
. ./test.sub1
echo "return status: $?"

echo "++ exit in a sourced script"
(. ./test.sub2;
 echo "this message should not be print"
)
echo "return status: $?"

