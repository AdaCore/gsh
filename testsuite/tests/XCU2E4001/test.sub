echo "+ value of \$1 in sourced script: '$1'"
toto="hello"
if [ "$1" = "" ]; then
   false
fi
