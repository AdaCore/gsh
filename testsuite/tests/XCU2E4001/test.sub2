echo "in test.sub2"
exit 42
