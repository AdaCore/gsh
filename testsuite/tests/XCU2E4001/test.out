++ ensure that a variable set in a sourced script is visible
+ value of $1 in sourced script: ''
return status: 1
$toto='hello'
++ use of source builtin without any argument
return status: 2
++ sourcing inexistant script
return status: 1
++ sourcing a script + positional arg
+ value of $1 in sourced script: 'a'
return status: 0
++ return in a sourced scripts
in test.sub1
return status: 1
++ exit in a sourced script
in test.sub2
return status: 42
